plugins/nominal-connection-checker/src/type_hierarchy.js

```diff
@@ -138,51 +178,64 @@ export class TypeHierarchy {
    * Theoretical Computer Science, 380.1-2 (2007): 37-46.
    * Link: https://bit.ly/2SrCRs5 .
    *
+   * But the above has been slightly modified to work for both ancestors and
+   * descendants.
+   *
    * Operates in O(nm) where n is the number of nodes and m is the number of
    * edges.
+   *
+   * @param {!Map<string, !Map<string, Array<string>>>} nearestCommonMap The
+   *      map of nearest common types (either ancestors or descendants) that we
+   *     are initializing.
+   * @param {function(TypeDef):!Array<string>} relevantRelatives Returns the
+   *     relatives that are relevant to this procedure. In the case of
+   *     ancestors, returns supertypes, and in the case of descendants, returns
+   *     subtypes.
+   * @param {function(TypeDef, string):boolean} isNearest Returns true if the
```

Comment:
I see. I originally thought that `subs()` was all subtypes including indirect and `descendants` was only direct subtypes. But after your explanation and looking at the jsDoc I see that it is the opposite.

So I don't think this function should be `isDirectRelative` now because that isn't the case (I wouldn't consider a to be a direct relative of c, I would only consider b to be a direct relative).

What's the motivation for calling it "nearest" in the first place?
When I read the description, I assumed that nearest == direct supertype/subtype (depending on which case).

I'm still thinking if I have a specific suggestion for a name.

Reply:
First: I reorganized the params/internal logic isNearest a bit so that it makes more sense in context (hopefully).

I originally called it nearest because this function returns true if the first type is the nearest common ancestor/descendant of itself and the second type. I think this makes more sense if you look at how it is used on L214. If it returns true, we can add the first type to the list of nearest common types between the two types.

Also note that by definition the second type can only be the nearest common ancestor/descendant of itsefl and the first type if the second type is the ancestor/descendant of the first (respectively). Which is why the anonymous functions passed to this function work.

Reply:
Thanks for the explanation and I think your reworded jsDoc for the `isNearest` param helps a lot.